src/persistent_log/mem.rs

```diff
@@ -86,18 +86,18 @@ impl Log for MemLog {
         }
     }
 
-    fn entry(&self, index: LogIndex) -> result::Result<(Term, &[u8]), Error> {
+    fn entry(&self, index: LogIndex) -> result::Result<(Term, Vec<u8>), Error> {
         let (term, ref bytes) = self.entries[(index - 1).as_u64() as usize];
-        Ok((term, &bytes))
+        Ok((term, bytes.to_owned()))
     }
 
     fn append_entries(&mut self,
                       from: LogIndex,
-                      entries: &[(Term, &[u8])])
+                      entries: &[(Term, Vec<u8>)])
```

Comment:
ok, doesn't seem like you could even get a flat array of entries out of the CapnProto encoding. So the only way to avoid an alloc here is to call `append_entries` for each entry individually, or to pass an iterator in instead of a slice. If anything, the latter sounds better.